src/java/org/apache/cassandra/dht/IPartitioner.java

```diff
@@ -128,8 +128,9 @@ default Token getMaximumToken()
     /**
      * Abstract type that orders the same way as DecoratedKeys provided by this partitioner.
      * Used by secondary indices.
+     * @param baseType base type for PartitionerDefinedOrder
      */
-    public AbstractType<?> partitionOrdering();
+    public AbstractType<?> partitionOrdering(AbstractType<?> baseType);
```

Comment:
A few nits:
- The `public` keyword is redundant
- The argument can be `null`
- The argument could be named in a more specific way (updating the JavaDoc comment, including a mention of when it's expected to be `null`)

```suggestion
   AbstractType<?> partitionOrdering(@Nullable AbstractType<?> partitionKeyType);
```